traits/tests/test_trait_list_object.py

```diff
@@ -0,0 +1,311 @@
+        # self.assertEqual(slice(1, 3, None), self.index)
+        self.assertEqual([], self.removed)
+        self.assertEqual([1, 2], self.added)
+
+    def test_insert(self):
+        tl = TraitList([2],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+
+        tl.insert(0, 1)  # [1,2]
+        self.assertEqual(0, self.index)
+        self.assertEqual([], self.removed)
+        self.assertEqual([1], self.added)
+
+        tl.insert(-1, 3)  # [1,3,2]
+        self.assertEqual(1, self.index)
+        self.assertEqual([], self.removed)
+        self.assertEqual([3], self.added)
+
+    def test_pop(self):
+        tl = TraitList([1, 2, 3, 4, 5],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+
+        tl.pop()
+        self.assertEqual(4, self.index)
+        self.assertEqual([5], self.removed)
+        self.assertEqual([], self.added)
+
+        tl.pop(0)
+        self.assertEqual(0, self.index)
+        self.assertEqual([1], self.removed)
+        self.assertEqual([], self.added)
+
+        # tl is now [2,3,4]
+        tl.pop(-2)
+        self.assertEqual(1, self.index)
+        self.assertEqual([3], self.removed)
+        self.assertEqual([], self.added)
+
+    def test_remove(self):
+        tl = TraitList([1, 2, 3, 4, 5],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+
+        tl.remove(3)
+        self.assertEqual(2, self.index)
+        self.assertEqual([3], self.removed)
+        self.assertEqual([], self.added)
+
+        with self.assertRaises(ValueError):
+            tl.remove(3)
+
+    def test_clear(self):
+        tl = TraitList([1, 2, 3, 4, 5],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+        tl.clear()
+        self.assertEqual(slice(0, 5, None), self.index)
+        self.assertEqual([1, 2, 3, 4, 5], self.removed)
+        self.assertEqual([], self.added)
+
+    def test_sort(self):
+        tl = TraitList([2, 3, 1, 4, 5, 0],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+
+        tl.sort()
+
+        self.assertEqual([0, 1, 2, 3, 4, 5], tl)
+        self.assertEqual(slice(0, 6, None), self.index)
+        self.assertEqual([2, 3, 1, 4, 5, 0], self.removed)
+        self.assertEqual([0, 1, 2, 3, 4, 5], self.added)
+
+    def test_reverse(self):
+        tl = TraitList([1, 2, 3, 4, 5],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+
+        tl.reverse()
+        self.assertEqual([5, 4, 3, 2, 1], tl)
+        self.assertEqual(slice(0, 5, None), self.index)
+        self.assertEqual([1, 2, 3, 4, 5], self.removed)
+        self.assertEqual([5, 4, 3, 2, 1], self.added)
+
+    def test_pickle(self):
+        tl = TraitList([1, 2, 3, 4, 5],
```

Comment:
Could we enhance this test to cover the various different pickle protocols available? I.e., try dumping with all available protocols from `0` to `pickle.HIGHEST_PROTOCOL` (inclusive).